src/node_util.cc

```diff
@@ -49,6 +49,35 @@ static void GetProxyDetails(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(ret);
 }
 
+static void PreviewEntries(const FunctionCallbackInfo<Value>& args) {
+  if (!args[0]->IsObject())
+    return;
+
+  bool is_key_value;
+  Local<Array> entries;
+  if (!args[0].As<Object>()->PreviewEntries(&is_key_value).ToLocal(&entries))
+    return;
+  if (!is_key_value)
+    return args.GetReturnValue().Set(entries);
+
+  uint32_t length = entries->Length();
+  CHECK_EQ(length % 2, 0);
```

Comment:
This check should actually be obsolete. We should realize it one way or the other if the V8 implementation is broken.

Reply:
Right now, if there’s a single extra element at the end, we wouldn’t notice that. Like, yes, obviously that would be a bug in V8, but I don’t think there’s much harm in the extra check?

Reply:
We should notice it because this code will only run in case we have a test for this code path and in those tests we actually strictly check the outcome.